internal/config/read_test.go

```diff
@@ -95,6 +99,57 @@ func TestConfigReadFromEnv_NotInteger(t *testing.T) {
 	}
 
 	if len(invConfigError) != 2 {
+		t.Errorf("%v", invConfigError)
 		t.Errorf("config.readFromEnv() error length = %d, want 2", len(invConfigError))
 	}
 }
+
+func TestConfigReadFromEnv_MissingBuildId(t *testing.T) {
+	os.Setenv("BUILDKITE_SPLITTER_BASE_URL", "")
+	os.Setenv("BUILDKITE_SPLITTER_MODE", "")
+	os.Setenv("BUILDKITE_SPLITTER_TEST_CMD", "")
+	os.Setenv("BUILDKITE_SPLITTER_RETRY_COUNT", "")
+	os.Setenv("BUILDKITE_STEP_ID", "123")
+	defer os.Clearenv()
+
+	c := Config{}
+	err := c.readFromEnv()
+
+	var invConfigError InvalidConfigError
+	if !errors.As(err, &invConfigError) {
+		t.Errorf("config.readFromEnv() error = %v, want InvalidConfigError", err)
+	}
+
+	want := "BUILDKITE_BUILD_ID must not be blank"
+
+	if errors.As(err, &invConfigError) {
+		if got := invConfigError[0].Error(); got != want {
+			t.Errorf("config.readFromEnv() got = %v, want = %v", got, want)
+		}
+	}
```

Comment:
[nit] we have the error struct conversion and check above, no need to check it again here.
```suggestion
	if got := invConfigError[0].Error(); got != want {
		t.Errorf("config.readFromEnv() got = %v, want = %v", got, want)
	}
```

Reply:
Ah yep, good catch, thanks!